### DISCUSSION TITLE
Put lsp config in multiple files

### DISCUSSION THREAD

**Discussion by neuromaancer** [LazyVim/LazyVim#2467] [Q&A] [answered]
I am a bit confused about the LSP config. I have a big file with all the LSP-related configs. How can I break it down it multiple files and a folder? What are your best practices?

**Comment by dpetka2001** [comment, marked as answer]
Not sure if it's best practice but here you can look at my dots about [lsp](https://github.com/dpetka2001/dotfiles/blob/main/dot_config/nvim/lua/plugins/lsp.lua). At the end of the file the important part is `{ import = "plugins.extras.lang" }`, which exposes another folder that I have in `plugins/extras/lang` with files per language and in each file corresponding settings about several plugins' specs.

**Reply by neuromaancer** [reply]
Thanks, so I just need to import a folder that contains multiple lsp_configs. I am wondering if it is possible just to break down the `opts.servers` into different files?

**Reply by dpetka2001** [reply]
It is possible. If you look inside the language files that I have, I have different `nvim-lspconfig` specs (with different `opts.servers` each) that in the end will all merge into a final one. This is `lazy.nvim`'s doing (the merging).

**Reply by neuromaancer** [reply]
Thanks it works!
